### DISCUSSION TITLE
Inconsistent filtered genes before and after scv.utils.merge function

### DISCUSSION THREAD

**Discussion by jipeifeng** [theislab/scvelo#471] [Q&A]
Hi:
    Thanks for the helpful tools.
    I have came across an issue at the preprocessing step, particularly at the filter_and_normalize function. 

   Test 1:
   I merged all the loom files and run the filter step:
  
<img width="979" alt="截屏2020-11-14 下午1 04 01" src="https://user-images.githubusercontent.com/24560774/99140137-e6340800-2679-11eb-8525-3f87e66f77ee.png">
  
  Test 2:
  I merged all the velocity output files (loom files),  merged the loom files into an already existing AnnData object, and run the filter step:

<img width="1040" alt="截屏2020-11-14 下午1 07 34" src="https://user-images.githubusercontent.com/24560774/99140190-5e9ac900-267a-11eb-8d2e-b723ac10de45.png">

And you can see that the number of filtered genes are inconsistent with each other. And would you please point what is the difference. Moreover, should I merge the Anndata object with scvelo object (using scv.utils.merge function) before or after performing the preprocessing step (scv.pp.filter_and_normalize).
Thank you very much !

**Comment by WeilerP** [comment]
You should first merge, then preprocess the data.

Regarding the different results after filtering: Let's say you detected gene A 5 times (in unspliced) in your first dataset but 0 times in the second. Suppose you are now filtering only based on unspliced counts via

```python
scv.filter_and_normalize(adata, min_counts_u=5)
```

Filtering the two datasets individually, gene A will not be filtered out in the first dataset. It is, however, filtered out in the second one. If you then merge the two datasets using an inner merge, you will not find gene A in the final AnnData object.
Merging the two datasets prior to filtering, will lead to gene A not being filtered as it is detected at least 5 times.
